concent_api/gatekeeper/views.py

```diff
@@ -0,0 +1,41 @@
+from base64                         import b64decode
+
+from django.conf                    import settings
+from django.http                    import JsonResponse, HttpResponse
+from django.views.decorators.http   import require_POST
+from django.views.decorators.csrf   import csrf_exempt
+
+from golem_messages.shortcuts       import load
+from golem_messages.message         import Message, init_messages
+
+
+@csrf_exempt
+@require_POST
+def file_transfer_auth(request):
+    init_messages()
+    if 'HTTP_AUTHORIZATION' not in request.META:
+        return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)
+
+    authorization = request.META['HTTP_AUTHORIZATION']
+
+    if len(authorization.split(" ")) != 2:
+        return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)
+    (golem_header, header_content) = authorization.split()
+
+    if golem_header != 'Golem':
+        return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)
+
+    try:
+        decoded_header_content = (b64decode(header_content))
+    except ValueError:
+        return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)
+
+    try:
+        check_golem_msg = load(decoded_header_content, settings.CONCENT_PRIVATE_KEY, settings.CONCENT_PUBLIC_KEY)
+    except AttributeError:
+        return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)
+
+    if isinstance(check_golem_msg, Message):
```

Comment:
How about switching it to 
```
if not isinstance(check_golem_msg, Message):
    return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)

return HttpResponse("", status=200)
```
which fits into code pattern above.

Reply:
I agree.

And actually I'd make this check more explicit. The result here can be only `None` or `Message`. `None` indicates a failure to decode the message. Anything else would be a bug. This assumption should be documented with `assert`:

``` python
if decoded_golem_message is None:
    return JsonResponse({"WWW-Authenticate": "Golem realm='Concent Storage'"}, status = 401)

assert isinstance(decoded_golem_message, Message)
return HttpResponse("", status = 200)
```

BTW, @Jakub89 please avoid naming variables like this. `check_golem_msg` sounds like a function name.

Reply:
@Jakub89 Still waiting for this change.